### PR TITLE
make Http2_RequestFailsWithAppropriateHttpProtocolException test more deterministic

### LINKED ISSUE #92647: [Test Failure] Http2_IncorrectServerPreface_RequestFailsWithAppropriateHttpProtocolException
## Build Information
Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=417421
Build error leg or test failing: System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.Http2_IncorrectServerPreface_RequestFailsWithAppropriateHttpProtocolException

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md#how-to-create-a-known-issue-step-by-step).

```json
{
  "ErrorMessage": "System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.Http2_IncorrectServerPreface_RequestFailsWithAppropriateHttpProtocolException [FAIL]",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=417421
**Error message validated:** `System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.Http2_IncorrectServerPreface_RequestFailsWithAppropriateHttpProtocolException [FAIL]`
**Result validation: :white_check_mark:** Known issue matched with the provided build.
**Validation performed at:** 9/26/2023 3:12:06 PM UTC

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[447249](https://dev.azure.com/dnceng-public/public/_build/results?buildId=447249)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_Http2.Http2_IncorrectServerPreface_RequestFailsWithAppropriateHttpProtocolException](https://dev.azure.com/dnceng-public/public/_build/results?buildId=447249&view=ms.vss-test-web.build-test-results-tab&runId=10011604&resultId=209349)|dotnet/runtime#86391|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|1|

### PR DESCRIPTION
**Pull request by wfurt**
When I look at the normal passing run I see something like

passingTest

when it fails, the exchange looks like 
failTest

So it seems like if the GOAWAY is sent before the client even start making request the test fails with  ` Couldn't find HttpProtocolException with matching error code in exception: System.Net.Http.HttpRequestException: An HTTP/2 connection could not be established because the server did not complete the HTTP/2 handshake.`

This may (or may not) be caused by the connection pool refactor. Since the requests are now detached, the connection can be closed by the 'GOAWAY` and the request does not fail with protocol error. 

To minimize risk, I only changed the test to wait for beginning of the request to arrive so when `HttpClient` process the response it would produce deterministic result. I was running tests in loop with this change and I did not see failure. It was failing for me before frequently with less than 20-50 iterations. 
 
fixes #92647

### CODE REVIEW

File: src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs [hunk 1]
```diff
@@ -139,7 +139,8 @@ public async Task Http2_StreamResetByServerBeforePrefix_RequestFailsWithGoawayPr
 
                 Http2LoopbackConnection connection = await server.AcceptConnectionAsync(timeout: null);
                 _ = await connection.ReadSettingsAsync();
-
+                // Wait until client starts sending request
+                Frame frame = await connection.ReadFrameAsync(TestHelper.PassingTestTimeout);
```
Comment (antonfirsov) [inline src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs:143]: Nit: we don't use the frame returned.

```suggestion
                _ = await connection.ReadFrameAsync(TestHelper.PassingTestTimeout);
```

File: src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs [hunk 2]
```diff
@@ -319,6 +320,8 @@ public async Task Http2_IncorrectServerPreface_RequestFailsWithAppropriateHttpPr
 
                 Http2LoopbackConnection connection = await server.AcceptConnectionAsync();
                 await connection.ReadSettingsAsync();
+                // Wait until client starts sending request
+                Frame frame = await connection.ReadFrameAsync(TestHelper.PassingTestTimeout);
```
Comment (antonfirsov) [inline src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs:324]:
```suggestion
                _ = await connection.ReadFrameAsync(TestHelper.PassingTestTimeout);
```

**Review by antonfirsov — APPROVED** [review]
LGTM

File: src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs (hunk 1 above)
Comment (wfurt) [reply src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs]: updated. I originally played with the frame type but I did not use it at the end.

**Comment by wfurt** [comment]
enterprised-linux failures are unrelated. looks like infrastructure problem.

**Pull request merged**